src/test/javascript/spec/component/account/ssh-user-settings-key-details.component.spec.ts

```diff
@@ -0,0 +1,163 @@
+                    useValue: new MockActivatedRoute({}),
+                },
+                { provide: AccountService, useValue: accountServiceMock },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: NgbModal, useClass: MockNgbModalService },
+                { provide: Router, useValue: routerMock },
+                { provide: AlertService, useValue: alertServiceMock },
+            ],
+        }).compileComponents();
+        fixture = TestBed.createComponent(SshUserSettingsKeyDetailsComponent);
+        comp = fixture.componentInstance;
+        translateService = TestBed.inject(TranslateService);
+        translateService.currentLang = 'en';
+
+        router = TestBed.inject(Router);
+        activatedRoute = TestBed.inject(ActivatedRoute) as unknown as MockActivatedRoute;
+    });
+
+    it('should initialize view for adding new keys and save new key', () => {
+        accountServiceMock.addNewSshPublicKey.mockReturnValue(of({}));
+        comp.ngOnInit();
+        expect(accountServiceMock.getSshPublicKey).not.toHaveBeenCalled();
+        expect(comp.isLoading).toBeFalse();
+        comp.displayedSshKey = mockKey;
+        comp.displayedExpiryDate = dayjs();
+        comp.displayedKeyLabel = 'label';
+        comp.validateExpiryDate();
+        expect(comp.isExpiryDateValid).toBeTrue();
+        comp.saveSshKey();
+        expect(alertServiceMock.success).toHaveBeenCalled();
+        expect(router.navigate).toHaveBeenCalledWith(['../'], { relativeTo: activatedRoute });
+    });
+
+    it('should initialize view for adding new keys and fail saving key', () => {
+        const httpError1 = new HttpErrorResponse({
+            error: { errorKey: invalidKeyFormat },
+            status: 400,
+        });
+        const httpError2 = new HttpErrorResponse({
+            error: { errorKey: keyAlreadyExists },
+            status: 400,
+        });
+        accountServiceMock.addNewSshPublicKey.mockReturnValue(throwError(() => httpError1));
+        comp.ngOnInit();
+        expect(accountServiceMock.getSshPublicKey).not.toHaveBeenCalled();
+        expect(comp.isLoading).toBeFalse();
+        comp.displayedSshKey = mockKey;
+        comp.displayedExpiryDate = dayjs();
+        comp.displayedKeyLabel = 'label';
+        comp.validateExpiryDate();
+        expect(comp.isExpiryDateValid).toBeTrue();
+        comp.saveSshKey();
+        accountServiceMock.addNewSshPublicKey.mockReturnValue(throwError(() => httpError2));
+        comp.saveSshKey();
+        expect(alertServiceMock.error).toHaveBeenCalled();
+        expect(router.navigate).not.toHaveBeenCalled();
+    });
+
+    it('should initialize key details view with key loaded', async () => {
+        accountServiceMock.getSshPublicKey.mockReturnValue(of(mockedUserSshKeys));
+        activatedRoute.setParameters({ keyId: 1 });
+        comp.ngOnInit();
+        expect(accountServiceMock.getSshPublicKey).toHaveBeenCalled();
```

Comment:
just a preference thing but not necessary, we could use more concrete testing methods like toHaveBeenCalledExactlyOnce()

Reply:
Might do that in a followup